clusterloader2/pkg/framework/framework.go

```diff
@@ -84,16 +88,47 @@ func (f *Framework) CreateAutomanagedNamespaces(namespaceCount int) error {
 	return nil
 }
 
+// ListAutomanagedNamespaces returns all existing automanged namespace names.
+func (f *Framework) ListAutomanagedNamespaces() ([]string, error) {
+	var automanagedNamespacesList []string
+	namespacesList, err := client.ListNamespaces(f.clientSet)
+	if err != nil {
+		return automanagedNamespacesList, err
+	}
+	for _, namespace := range namespacesList {
+		if isAutomanagedNamespace(namespace) {
+			automanagedNamespacesList = append(automanagedNamespacesList, namespace)
+		}
+	}
+	return automanagedNamespacesList, nil
+}
+
 // DeleteAutomanagedNamespaces deletes all automanged namespaces.
-func (f *Framework) DeleteAutomanagedNamespaces() error {
+func (f *Framework) DeleteAutomanagedNamespaces() []error {
+	var wg wait.Group
+	var lock sync.Mutex
+	var errList []error
 	for i := 1; i <= f.automanagedNamespaceCount; i++ {
 		name := fmt.Sprintf("%v-%d", AutomanagedNamespaceName, i)
-		if err := client.DeleteNamespace(f.clientSet, name); err != nil {
-			return err
-		}
+		wg.Start(func() {
+			if err := client.DeleteNamespace(f.clientSet, name); err != nil {
+				lock.Lock()
+				defer lock.Unlock()
+				errList = append(errList, err)
+				return
+			}
+			if err := client.WaitForDeleteNamespace(f.clientSet, name); err != nil {
+				lock.Lock()
+				defer lock.Unlock()
+				errList = append(errList, err)
+			}
+		})
 	}
-	f.automanagedNamespaceCount = 0
-	return nil
+	wg.Wait()
+	if len(errList) < 1 {
+		f.automanagedNamespaceCount = 0
```

Comment:
Why do you need to do this?

Reply:
This represents the state of the cluster. For example, if `CreateAutomanagedNamespaces` is run when not every automanaged namespace is removed, it will return error without trying to create the namespaces.

Reply:
Lines removed.